lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -86,37 +93,70 @@ define( (require, exports, module) ->
     meta: 'equation'
   )
   
-  init = (editor, bindKey) ->
-    HashHandler = require("ace/keyboard/hash_handler").HashHandler; 
-    keyboardHandler = new HashHandler();
+  # Specific for token's system of type in ace
+  # We saw such a realization in html_completions.js
+  istype = (token, type) -> 
+    return token.type.lastIndexOf(type) > -1
+  
+
+  init = (editor, bindKey) -> 
+    HashHandler = require("ace/keyboard/hash_handler").HashHandler
+    keyboardHandler = new HashHandler()
     keyboardHandler.addCommand(
-      name: 'add item in list mode',
-      bindKey: bindKey,
-      exec: (editor) ->
+      name: 'add item in list mode'
+      bindKey: bindKey
+      exec: (editor) ->  
         pos = editor.getCursorPosition()
-        curLine = editor.session.getLine(pos.row);
+        curLine = editor.session.getLine(pos.row)
+        indentCount = ContextHelper.getNestedListDepth(editor.session, pos.row)
+        tabString = editor.getSession().getTabString()
         # it's temporary fix bug with added \item before \begin{itemize|enumerate}
-        if (editor.getSession().getContext(pos.row) == "list" && curLine.indexOf("begin") < pos.column) 
-          editor.insert("\n\t\\item ")
+        if ContextHelper.getContext(editor.session, pos.row) == LIST_STATE && curLine.indexOf("begin") < pos.column
+          editor.insert("\n" + tabString.repeat(indentCount) + "\\item ")
           return true
         else 
-          return false
+          return false   
```

Comment:
I recommend switching on  "trim trailing whitespace" option in your editor, otherwise we'll have lots of such rubbish diffs